### DISCUSSION TITLE
Windows Defender blocked starship.exe

### DISCUSSION THREAD

**Discussion by realmorrisliu** [starship/starship#1412] [General]
It is said that Windows Defender has detected Trojan:Win32/Occamy.C72 in starship.exe which i was installed by scoop
![image](https://user-images.githubusercontent.com/8832717/86083384-a5719300-bacc-11ea-8779-7620cb53314c.png)

**Comment by chipbuster** [comment]
Can confirm that I see the same behavior on my copy of Windows.

However, things are a bit strange: if I attempt to scan the file itself, nothing happens (windows defender says the file is clean). If I attempt to run the file, it gets blocked by Windows Defender. Uploading the file to a virus scanner results in mostly nonmalicious results, with [the filenames its seen as](https://www.virustotal.com/gui/file/cb440b8a08a2095a59666a859b35aa5a1524b140b909ecc760f38f3baccf80e6/details) corresponding to common CLI utilities. The SHA256-hash of the file is constant throughout this test.

Finally, all the executables in my `~/scoop/shim` directory are all of the exact same size, about 8kB. Since starship itself is about 2MB, this makes me fairly certain that these aren't actually the executables downloaded by scoop. Rather, they appear to be some sort of executable adapter that's responsible for launching the actual program. Unfortunately, I don't know enough about scoop, Defender, or Windows at large to be able to say exactly what's going on.

Since this is apparently an issue [that has occurred in the past](https://github.com/lukesampson/scoop/issues/3175) with scoop, I'm inclined to lean towards "behavioral false positive" at the moment, though again I can't really be sure since I don't know enough about the systems involved.

**Reply by realmorrisliu** [reply]
Thanks for the reply, I'll just allow the executable and restore it

**Reply by r-darwish** [reply]
I am also experiencing it with Scoop. However, I also happens when I download the zip file manually from Github:

![image](https://user-images.githubusercontent.com/1383193/86089930-d5f80380-bab1-11ea-8afc-bb5084da7d50.png)

It seems consistent with 0.43.0 but previous versions are not detected as threats

**Reply by chipbuster** [reply]
Hmm, that's not excellent...does it show up under defender if you scan the file as well or only if you execute it?

**Reply by r-darwish** [reply]
Also when scanning the zip file manually:
![image](https://user-images.githubusercontent.com/1383193/86091786-26249500-bab5-11ea-88e4-23c53107033c.png)

**Reply by chipbuster** [reply]
Okay, rebooted into Windows and I see it too.

Do you think this is something to be concerned about? If there's even a bit of doubt, I think I can just pull the Windows binaries until tomorrow.

**Reply by r-darwish** [reply]
I'm not sure. Perhaps it's best to be on the safe side. I wonder how deterministic Rust binaries are. If they are, I can compile starship on my machine and we'll compare the binaries

**Reply by chipbuster** [reply]
I'm doing that right now as well. Very well, I'll err on the safe side and pull the Windows binary for tonight--hopefully we can have it back up by tomorrow.

**Reply by chipbuster** [reply]
@r-darwish Apparently the answer is "not very"...

**Reply by r-darwish** [reply]
[starship.zip](https://github.com/starship/starship/files/4850080/starship.zip)

This is `cargo build --release` from my machine. I built the tag `v0.43.0`. I tested it on another Windows machine. It does not get recognized as a virus.

**Reply by chipbuster** [reply]
Our file hashes are different. 

```
PS C:\Users\chipb\tmp\starship\target\release> Get-FileHash .\starship.exe
Algorithm       Hash                                                                   Path
---------       ----                                                                   ----
SHA256          90884B3941788B0E104BCAF33C0ACA453615C4477C76704A4F15EE4E2472BF99       C:\Users\chipb\tmp\starship\target\release...

PS C:\Users\chipb\tmp\starship\target\release> Get-FileHash .\starship-darwish.exe

Algorithm       Hash                                                                   Path
---------       ----                                                                   ----
SHA256          FA0697AD67905DE170ED51C5F764FB2F85D201F35A818CA14207859C23006D8D       C:\Users\chipb\tmp\starship\target\release...

```

Here's the binary from `cargo build --release` on mine, tag v0.43.0, using `rustc 1.44.1 (c7087fe00 2020-06-17)

[starship.zip](https://github.com/starship/starship/files/4850133/starship.zip)

Unfortunately, it's late enough and I've been working long enough that I can no longer trust my judgement. If you'd like to keep poking at this, feel free to let me know anything you want from me, and I'll send it once I wake up.

**Reply by r-darwish** [reply]
Thanks for helping out. I'll update you if I find anything interesting.

**Comment by matchai** [comment]
The binary from the latest Starship release has been submitted to Microsoft for analysis. I'll post here with any analyst comments.

**Reply by matchai** [reply]
Microsoft has confirmed that it was a false-positive and has cleared the detection.
If you see the v0.43.0 release flagged by Windows Defender, please follow these steps to clear your cached detection and obtain the latest malware definitions:

1. Open command prompt as administrator and change directory to c:\Program Files\Windows Defender
2. Run “MpCmdRun.exe -removedefinitions -dynamicsignatures”
3. Run "MpCmdRun.exe -SignatureUpdate"

Alternatively, the latest definition is available for download here: https://www.microsoft.com/en-us/wdsi/definitions

**Reply by realmorrisliu** [reply]
Updated Windows Defender definition from Windows Update, works great👍
